messaging/src/main/java/org/springframework/security/messaging/context/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -133,26 +149,63 @@ public void setSecurityContextHolderStrategy(SecurityContextHolderStrategy secur
 		this.securityContextHolderStrategy = securityContextHolderStrategy;
 	}
 
+	/**
+	 * Configure AuthenticationPrincipal template resolution
+	 * <p>
+	 * By default, this value is <code>null</code>, which indicates that templates should
+	 * not be resolved.
+	 * @param defaults - whether to resolve AuthenticationPrincipal templates parameters
+	 * @since 6.4
+	 */
+	public void setTemplateDefaults(@Nullable PrePostTemplateDefaults defaults) {
+		if (defaults != null) {
```

Comment:
Please instead don't allow null values. We don't want methods to silently not do what they say they will do.

Reply:
Done